### PR TITLE
Simplify boolean expressions

### PR DESCRIPTION
**Pull request by simon04**

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/Column.java [hunk 1]
```diff
@@ -214,7 +214,7 @@ void updateValue(JDBCType jdbcType, Object value, JavaType javaType, StreamSette
                 if (JDBCType.TINYINT == cryptoMetadata.getBaseTypeInfo().getSSType().getJDBCType()
                         && javaType == JavaType.SHORT) {
                     if (value instanceof Boolean) {
-                        if (true == ((boolean) value)) {
+                        if (((boolean) value)) {
```
Comment (rene-ye) [inline src/main/java/com/microsoft/sqlserver/jdbc/Column.java]: Unnecessary brackets around `(boolean) value`.

File: src/main/java/com/microsoft/sqlserver/jdbc/Parameter.java [hunk 2]
```diff
@@ -296,7 +296,7 @@ void setValue(JDBCType jdbcType, Object value, JavaType javaType, StreamSetterAr
         }
 
         // forceEncryption is true, shouldhonorae is false
-        if ((true == forceEncrypt) && (false == Util.shouldHonorAEForParameters(stmtColumnEncriptionSetting, con))) {
+        if ((forceEncrypt) && (!Util.shouldHonorAEForParameters(stmtColumnEncriptionSetting, con))) {
```
Comment (rene-ye) [inline src/main/java/com/microsoft/sqlserver/jdbc/Parameter.java]: Unnecessary brackets around both `forceEncrypt` and `!Util.shouldHonorAEForParameters(stmtColumnEncriptionSetting, con)`. It seems the variable also has a typo as it should be `stmtColumnEncryptionSetting`, but the team will address this in another PR.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java [hunk 3]
```diff
@@ -1529,7 +1529,7 @@ Connection connectInternal(Properties propsIn,
             }
             authenticationString = SqlAuthentication.valueOfString(sPropValue).toString();
 
-            if ((true == integratedSecurity)
+            if ((integratedSecurity)
```
Comment (rene-ye) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java]: Unnecessary brackets around `integratedSecurity` and `!authenticationString.equalsIgnoreCase(SqlAuthentication.NotSpecified.toString()`

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java [hunk 4]
```diff
@@ -2983,7 +2983,7 @@ String sqlStatementToSetTransactionIsolationLevel() throws SQLServerException {
      * @return the required syntax
      */
     static String sqlStatementToSetCommit(boolean autoCommit) {
-        return (true == autoCommit) ? "set implicit_transactions off " : "set implicit_transactions on ";
+        return (autoCommit) ? "set implicit_transactions off " : "set implicit_transactions on ";
```
Comment (rene-ye) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java]: Unnecessary brackets around `autoCommit`

File: src/main/java/com/microsoft/sqlserver/jdbc/Util.java [hunk 5]
```diff
@@ -487,9 +487,9 @@ else if (ch == ':')
                 name = SQLServerDriver.getNormalizedPropertyName(name, logger);
                 if (null != name) {
                     if (logger.isLoggable(Level.FINE)) {
-                        if ((false == name.equals(SQLServerDriverStringProperty.USER.toString()))
-                                && (false == name.equals(SQLServerDriverStringProperty.PASSWORD.toString()))
-                                && (false == name.equals(SQLServerDriverStringProperty.KEY_STORE_SECRET.toString())))
+                        if ((!name.equals(SQLServerDriverStringProperty.USER.toString()))
```
Comment (rene-ye) [inline src/main/java/com/microsoft/sqlserver/jdbc/Util.java]: Unnecessary brackets around `!name.equals(SQLServerDriverStringProperty.USER.toString()`, `!name.equals(SQLServerDriverStringProperty.PASSWORD.toString()`, and `!name.equals(SQLServerDriverStringProperty.KEY_STORE_SECRET.toString()`

File: src/main/java/com/microsoft/sqlserver/jdbc/Util.java [hunk 6]
```diff
@@ -417,8 +417,8 @@ else if (ch == ':')
                         name = SQLServerDriver.getNormalizedPropertyName(name, logger);
                         if (null != name) {
                             if (logger.isLoggable(Level.FINE)) {
-                                if ((false == name.equals(SQLServerDriverStringProperty.USER.toString()))
-                                        && (false == name.equals(SQLServerDriverStringProperty.PASSWORD.toString())))
+                                if ((!name.equals(SQLServerDriverStringProperty.USER.toString()))
```
Comment (rene-ye) [inline src/main/java/com/microsoft/sqlserver/jdbc/Util.java]: Unnecessary brackets around both `!name.equals(SQLServerDriverStringProperty.USER.toString()` and `!name.equals(SQLServerDriverStringProperty.PASSWORD.toString()`

**Comment by simon04** [comment]
@rene-ye, thanks a lot for your detailed and applicable review. I addressed your suggestions and amended+rebased my commit.

**Pull request merged**